### PR TITLE
fix: databricks retries

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

- Add `databricks` retries policy to the `databricks` client. This is basically used as a cold start time in order to connect to the `databricks` cluster.

## Notion Ticket

https://www.notion.so/rudderstacks/Databricks-intermittent-failures-d5e939f3749543888b8f070a4c14c7e2?pvs=4

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by cisse21 — APPROVED** [review]

File: warehouse/integrations/deltalake-native/deltalake.go [hunk 1]
```diff
@@ -147,6 +150,9 @@ func WithConfig(h *Deltalake, config *config.Config) {
 	h.LoadTableStrategy = config.GetString("Warehouse.deltalake.loadTableStrategy", mergeMode)
 	h.EnablePartitionPruning = config.GetBool("Warehouse.deltalake.enablePartitionPruning", true)
 	h.SlowQueryThreshold = config.GetDuration("Warehouse.deltalake.slowQueryThreshold", 5, time.Minute)
+	h.Retries = config.GetInt("Warehouse.deltalake.retries", 10)
+	h.RetryMaxWait = config.GetDuration("Warehouse.deltalake.retryMaxWait", 1, time.Second)
+	h.RetryMinWait = config.GetDuration("Warehouse.deltalake.retryMinWait", 300, time.Second)
```
Comment (fracasula) [inline warehouse/integrations/deltalake-native/deltalake.go:155]: Shouldn't it be min wait 1s and max wait 300s? :thinking:

File: warehouse/integrations/deltalake-native/deltalake.go [hunk 2]
```diff
@@ -191,6 +197,7 @@ func (d *Deltalake) connect() (*sqlmiddleware.DB, error) {
 			warehouseutils.GetConfigValue(catalog, d.Warehouse),
 			"",
 		),
+		dbsql.WithRetries(d.Retries, d.RetryMaxWait, d.RetryMinWait),
```
Comment (fracasula) [inline warehouse/integrations/deltalake-native/deltalake.go:200]: These seem to be mixed up as well, the 2nd parameter should be min wait and the 3rd max wait.

```go
func WithRetries(retryMax int, retryWaitMin time.Duration, retryWaitMax time.Duration) connOption {
	return func(c *config.Config) {
		c.RetryWaitMax = retryWaitMax
		c.RetryWaitMin = retryWaitMin
		c.RetryMax = retryMax
	}
}
```

Comment (fracasula) [reply warehouse/integrations/deltalake-native/deltalake.go:200]:
```suggestion
		dbsql.WithRetries(d.Retries, d.RetryMinWait, d.RetryMaxWait),
```

File: warehouse/integrations/deltalake-native/deltalake.go [hunk 3]
```diff
@@ -134,6 +134,9 @@ type Deltalake struct {
 	LoadTableStrategy      string
 	EnablePartitionPruning bool
 	SlowQueryThreshold     time.Duration
+	Retries                int
```
Comment (fracasula) [inline warehouse/integrations/deltalake-native/deltalake.go:137]: Can we rename this to `MaxRetries` please?

**Review by fracasula — APPROVED** [review]
I left a suggestion to rename `Retries` to `MaxRetries` since it makes more sense in my opinion. The rest looks good, thanks @achettyiitr

**Comment by achettyiitr** [comment]
Thanks, @fracasula. Addressed it.

**Review by lvrach — APPROVED** [review]

**Pull request merged**